examples/jsm/webxr/XRHandPrimitiveModel.js

```diff
@@ -12,93 +13,84 @@ class XRHandPrimitiveModel {
 
 		this.controller = controller;
 		this.handModel = handModel;
+		this.envMap = null;
 
-	  this.envMap = null;
+		let geometry;
+		const jointMaterial = new MeshStandardMaterial( { color: 0xffffff, roughness: 1, metalness: 0 } );
 
-		this.handMesh = new Group();
-		this.handModel.add( this.handMesh );
-
-		if ( window.XRHand ) {
-
-			let geometry;
-
-			if ( ! options || ! options.primitive || options.primitive === 'sphere' ) {
+		if ( ! options || ! options.primitive || options.primitive === 'sphere' ) {
 
-				geometry = new SphereGeometry( 1, 10, 10 );
+			geometry = new SphereGeometry( 1, 10, 10 );
 
-			} else if ( options.primitive === 'box' ) {
+		} else if ( options.primitive === 'box' ) {
 
-				geometry = new BoxGeometry( 1, 1, 1 );
+			geometry = new BoxGeometry( 1, 1, 1 );
 
-			}
-
-			const jointMaterial = new MeshStandardMaterial( { color: 0xffffff, roughness: 1, metalness: 0 } );
-			const tipMaterial = new MeshStandardMaterial( { color: 0x999999, roughness: 1, metalness: 0 } );
-
-			const joints = [
-				'wrist',
-				'thumb-metacarpal',
-				'thumb-phalanx-proximal',
-				'thumb-phalanx-distal',
-				'thumb-tip',
-				'index-finger-metacarpal',
-				'index-finger-phalanx-proximal',
-				'index-finger-phalanx-intermediate',
-				'index-finger-phalanx-distal',
-				'index-finger-tip',
-				'middle-finger-metacarpal',
-				'middle-finger-phalanx-proximal',
-				'middle-finger-phalanx-intermediate',
-				'middle-finger-phalanx-distal',
-				'middle-finger-tip',
-				'ring-finger-metacarpal',
-				'ring-finger-phalanx-proximal',
-				'ring-finger-phalanx-intermediate',
-				'ring-finger-phalanx-distal',
-				'ring-finger-tip',
-				'pinky-finger-metacarpal',
-				'pinky-finger-phalanx-proximal',
-				'pinky-finger-phalanx-intermediate',
-				'pinky-finger-phalanx-distal',
-				'pinky-finger-tip'
-			];
-
-			for ( const jointName of joints ) {
-
-				var cube = new Mesh( geometry, jointName.indexOf( 'tip' ) !== - 1 ? tipMaterial : jointMaterial );
-				cube.castShadow = true;
-				cube.receiveShadow = true;
-				cube.jointName = jointName;
-				this.handMesh.add( cube );
+		}
 
-			}
+		this.handMesh = new InstancedMesh( geometry, jointMaterial, 30 );
```

Comment:
I honestly do not know how much this does, but I guess it would be good practice to add this?

```js
this.handMesh.instanceMatrix.setUsage( THREE.DynamicDrawUsage ); // will be updated every frame
```

Reply:
I didn't know about this one; that does seem appropriate here.

Thank you for letting me know about it!

Reply:
It's still a mystery to me 😁